src/Akka.HealthCheck/Configuration/akka.healthcheck.conf

```diff
@@ -11,13 +11,24 @@ akka.healthcheck{
 	# Such as Liveness/Readiness subscriptions, and status request
 	log-info = on
 
-	liveness{
-		# The default IProbeProvider implementation used for executing
-		# liveness checks inside Akka.Healtcheck. Can be overridden by
-		# end-users via a custom IProbeProvider implementation. 
-		#
-		# Supply your fully qualified name (FQN) here to override it.
-		provider = "Akka.HealthCheck.Liveness.DefaultLivenessProvider, Akka.HealthCheck"
+	liveness {
+	    # List of liveness probe providers. 
+	    # Custom end-user provider can be created by implementing the IProbeProvider interface.
+	    providers {
```

Comment:
Got it, that makes a lot of sense

Reply:
All of the probe providers and probe actors themselves are exposed as an ImmutableDictionary inside in the `HealthCheck` extension, each are keyed to the property name inside `akka.healthcheck.liveness.providers` or `akka.healthcheck.readiness.providers`.

To override an implementation, an end user need to implement the `IProbeProvider` interface, then assign it to the name they wanted. For example, to override the default provider, the user can set `akka.healthcheck.readiness.providers.default = "Custom.Package.ProbeProvider, Custom.Package"`, or even have their custom probe living side by side with the default one:

```hocon
akka.healthcheck.readiness.providers {
  default = "Akka.HealthCheck.Readiness.DefaultReadinessProvider, Akka.HealthCheck"
  custom = "Custom.Package.ProbeProvider, Custom.Package"
}
```

The "custom" provider can be accessed by using
```csharp
var provider = HealthCheck.Get(system).ReadinessProviders["custom"];
```

The "custom" probe actor can be accessed by using
```csharp
var readinessActor = HealthCheck.Get(system).ReadinessProbes["custom"]
```

same thing can be done with the default provider.